### PR TITLE
Add API for reading libvmi.conf and returning GHashTable

### PR DESCRIPTION
**Pull request by skvl**

### CODE REVIEW

**Comment by skvl** [comment]
There was no obvious way to parse _libvmi.conf_ and return `GHasTable` for that.

Using _libvmi.conf_ allows to save some time on initialization while skipping most of _libvmi/os/windows/core.c:init_from_rekall_profile()_.

More over sometimes `windows_init()` fails because some symbols are not reachable (may be paged out, I don't know). If one could create _libvmi.conf_ and take snapshot of _VM_ than it is possible to reliably attach to the restored _VMs_.

I hope I succeed to explain the motivation behind the work despite of my bad english.

File: libvmi/libvmi.h [hunk 1]
```diff
@@ -741,6 +741,20 @@ page_mode_t vmi_init_paging(
...
+ *
+ * @param[in] vmi LibVMI instance
+ * @param[in,out] config Return GHashTable with parsed data
+ * @param[out] error Optional. If not NULL and the function returns VMI_FAILURE,
+ *                   this will specify the reason of failure.
+ * @return VMI_SUCCESS or VMI_FAILURE
+ */
+status_t vmi_read_config_file(
+    vmi_instance_t vmi,
+    void **config,
```
[the remark this answers is not in the document]
Comment (skvl) [reply libvmi/libvmi.h]: Thank You. I will replace it.

**Comment by skvl** [comment]
Once more I have added _expamples/win-offsets.c_ to help with creation of _libvmi.conf_.

File: examples/win-offsets.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define _GNU_SOURCE
+
+#include <libvmi/libvmi.h>
+#include <libvmi/peparse.h>
+#include <libvmi/os/windows/windows.h>
```
Comment (tklengyel) [inline examples/win-offsets.c]: This header is not installed and is internal to LibVMI only. You can't use it in an example or external project.

**Comment by tklengyel** [comment]
> There was no obvious way to parse libvmi.conf and return GHasTable for that.

There are accessor functions for every piece of information that is defined in the configuration file already. You can query each and store it in whatever datastructure you want, including a GHashTable.

> Using libvmi.conf allows to save some time on initialization while skipping most of libvmi/os/windows/core.c:init_from_rekall_profile().

I'm not sure what you mean by that. Why would that be skipped?

> More over sometimes windows_init() fails because some symbols are not reachable (may be paged out, I don't know). If one could create libvmi.conf and take snapshot of VM than it is possible to reliably attach to the restored VMs.

It might be worth investigating more what symbols are missing and why that is. I'm not convinced this PR represents a real fix to this potential scenario.

**Comment by skvl** [comment]
> I'm not sure what you mean by that. Why would that be skipped?
In `libvmi/core.c:vmi_init_os()` one could pass `VMI_CONFIG_GLOBAL_FILE_ENTRY` to read _libvmi.conf_ but this will not update function argument `config`. Or one could pass `VMI_CONFIG_GHASHTABLE` but this will not result in parsing _libvmi.conf_.

> There are accessor functions for every piece of information ...
How to get all this info?
```
win_ntoskrnl:0x2864000
win_ntoskrnl_va:0xfffff80002864000
win_tasks:0x188
win_pdbase:0x28
win_pid:0x180
win_pname:0x2e0
win_kdvb:0x0
win_sysproc:0xfffffa800172a040
win_kpcr:0x0
win_kdbg:0x0
rekall_profile:"/home/skovalev/images/7-sp1-x64/rekall-kernel.json"
ostype:"Windows"
```

> It might be worth investigating more what symbols are missing and why that is.
I will as soon as possible. This is hard to reproduce.

**Comment by skvl** [comment]
OK. I have found to achieve this in another way. So close the PR. Thank You very much for heading me to this.

**Pull request closed without merging**